builtin/logical/pki/backend.go

```diff
@@ -318,6 +333,14 @@ func (b *backend) initialize(ctx context.Context, _ *logical.InitializationReque
 	return nil
 }
 
+func setOcspStatus(b *backend, ctx context.Context) {
```

Comment:
On second thought, I'm not sure I like this pattern. I think I like the pattern we used in the migration invalidate func, wherein we set a flag to redo the read from the periodic func, using a better storage reference.

The actual setter could take an existing config (e.g., from the update handler or from this other context) and the periodic func would fetch the storage (or, delegate to another handler to fetch from storage) when an atomic flag to reload stuff is triggered from the invalid func.

For auto-rebuilding of CRLs, we'll want to reuse the handler (`setOcspAndCRLStatus` or something...Names tbd) and update these values too.

My 2c.

Reply:
I'm inclined to address this in my auto-rebuild PR.

Reply:
https://github.com/hashicorp/vault/pull/16762 -- added this here, let me know your thoughts on the approach @stevendpclark  :-)

Reply:
Yup that is a better pattern, will add some feedback on that PR.